source/main/windows/windowBounds.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import { debounce } from 'lodash-es/function';
+
+interface Rectangle {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+interface Window {
+  getBounds(): Rectangle;
+  on(event: string, handler: Function): void;
+}
+
+interface Display {
+  workArea: Rectangle;
+}
+
+interface Screen {
+  getDisplayMatching(rect: Rectangle): Display;
+  workArea: Rectangle;
+}
+
+interface Store {
+  set: (key: string, value: Rectangle) => void;
+  get: (key: string) => ?Rectangle;
+}
+
+const windowBoundsDefaultStoreKey = 'windowBounds';
+
+function getRightEdge(rect: Rectangle) {
+  return rect.x + rect.width;
+}
+
+function getBottomEdge(rect: Rectangle) {
+  return rect.y + rect.height;
+}
+
+function isWithinBounds(rect: Rectangle, bounds: Rectangle): boolean {
+  return (
+    rect.x >= bounds.x &&
+    rect.y >= bounds.y &&
+    getRightEdge(rect) <= getRightEdge(bounds) &&
+    getBottomEdge(rect) <= getBottomEdge(bounds)
+  );
+}
+
+function isFittingIntoBounds(rect: Rectangle, bounds: Rectangle): boolean {
+  return rect.width <= bounds.width && rect.height <= bounds.height;
+}
+
+function getCenteredRectInBounds(
+  rect: Rectangle,
+  bounds: Rectangle
+): Rectangle {
+  return {
+    width: rect.width,
+    height: rect.height,
+    x: bounds.x + (bounds.width - rect.width) / 2,
+    y: bounds.y + (bounds.height - rect.height) / 2,
+  };
+}
+
+export function saveWindowBounds(
```

Comment:
Not sure, but do you really need this function? You are only mutating the store.🤔

Reply:
Nice catch, will remove it - this is an artifact from the early stages which is not needed indeed 👍